prometheus.lua

```diff
@@ -536,7 +535,7 @@ function Prometheus:collect()
       self:log_error("Error getting '", key, "': ", err)
     end
   end
-  ngx.print(output)
+  return table.concat(output, "")
```

Comment:
Probably there is another way, we can add some new parameters, something like:
`Prometheus:collect()` - for default behavior 

And some configurable call mode

`
local set_headers = False
local return_expected = False
Prometheus:collect(set_headers, return_expected)
`

Looks like this will keep backwards compatibility and will add stream support transparately

Reply:
I'd suggest to keep it simple and just refactor existing code in a way that makes it possible to get the values as a table by calling another function.